### ISSUE TITLE
Property "enableGeneratedCatalog" has not been set by default

### ISSUE THREAD

**Issue by aharbunou** [googleads/google-ads-java#224]
Hi,

I've just upgraded to latest 4.4.0 version to verify the fix https://github.com/googleads/google-ads-java/issues/194.
Now I have different error saying "Property "enableGeneratedCatalog" has not been set".
Is "enableGeneratedCatalog" property required to set now? I didn't find any details about this in release notes for the new version. 
Could you please make it optional with some default value (like false) to keep the same behavior as it was before? 

Thanks,
Andrei

**Comment by jradcliff** [comment]
Hi Andrei,

I just ran a quick test on my side without the `enableGeneratedCatalog` property set, and I was able to create a `GoogleAdsClient`, get a service client, and issue a request without this error, so I need some more details to investigate.

Could you provide a stack trace so I can see precisely where this is failing? It would also be helpful to see your code that leads to the exception so I can reproduce the issue on my side.

Thanks,
Josh, Google Ads API Team

**Comment by aharbunou** [comment]
Please find the test below. It doesn't work for 4.3.0+ versions. 

```
package test.google.lib

import com.google.ads.googleads.lib.GoogleAdsClient
import com.google.auth.oauth2.UserCredentials
import org.junit.jupiter.api.Test

class GoogleLibTest {

    @Test
    fun `should build client`() {
        val creds = UserCredentials.newBuilder()
                .setClientId("1")
                .setClientSecret("2")
                .setRefreshToken("3")
                .build()

        val googleAdsClient = GoogleAdsClient.newBuilder()
                .setCredentials(creds)
                .setDeveloperToken("4")

       googleAdsClient.build()
    }
}
```

The error is the following.

```
java.lang.IllegalStateException: Property "enableGeneratedCatalog" has not been set

	at com.google.ads.googleads.lib.AutoValue_GoogleAdsClient$Builder.getEnableGeneratedCatalog(AutoValue_GoogleAdsClient.java:220)
	at com.google.ads.googleads.lib.GoogleAdsClient$Builder.build(GoogleAdsClient.java:308)
	at test.google.lib.GoogleLibTest.should build client(GoogleLibTest.kt:21)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:515)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:171)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:167)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:114)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:59)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:131)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:107)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:107)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
```

Thanks,
Andrei

**Comment by jradcliff** [comment]
Hi Andrei,

Thank you for the snippet. I just realized why our tests didn't catch this: we parameterized them to test the `true` and `false` cases, but didn't have a test for when `enableGeneratedCatalog` wasn't set at all. I've prepared [a pull request](https://github.com/googleads/google-ads-java/pull/225) with a fix and a new test.

Best regards,
Josh, Google Ads API Team

**Comment by jradcliff** [comment]
Hi Andrei,

I just released [v4.4.1](https://github.com/googleads/google-ads-java/releases/tag/4.4.1) with a fix for this issue. Please give it a try and let me know if you encounter any problems.

Thanks,
Josh, Google Ads API Team
